### PR TITLE
Add ClusterTier to cluster create parameters

### PR DESCRIPTION
**Pull request by ahosnyms**
Cluster tier supports premium and standard cluster types.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@ahosnyms__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ahosny). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by ahosnyms** [comment]
AzureSDKReviewers, do you have an estimate to merge this PR?

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

File: src/ResourceManagement/HDInsight/HDInsight.Tests/Helpers/HDInsightManagementTestUtilities.cs [hunk 1]
```diff
@@ -57,5 +57,28 @@ public static string CreateResourceGroup(ResourceManagementClient resourcesClien
...
+            {
+                var cluster = hdInsightClient.Clusters.Get(resourceGroup, dnsName);
+
+                if (cluster.Cluster.Properties.ClusterState == "Error")
+                {
+                    createError = true;
+                    break;
+                }
+                if (cluster.Cluster.Properties.ClusterState == "Running") { return; }
+                System.Threading.Thread.Sleep(2000);
```
Comment (hovsepm) [inline src/ResourceManagement/HDInsight/HDInsight.Tests/Helpers/HDInsightManagementTestUtilities.cs]: @ahosnyms use TestMockSupport.Delay(2000) instead

**Pull request merged**